### PR TITLE
Functional Tests -  Add monitoring products test

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add new functional test : monitoring products test
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `TEST_PATH="functional/BO/catalog/monitoring/02_monitoringProducts.js" URL_FO=SHOP_URL npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16905)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    },
+    {
+      args: {
+        productType: 'without price',
+        productToCreate: productWithoutPrice,
+        gridName: 'product_without_price',
+        enabled: true,
+      },
+    },
+  ];
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:103]:
```suggestion
  ];
  
```

Pleaaase some space in the code, it increase lisibility !

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  tests.forEach((test) => {
+    describe(`Create product ${test.args.productType} in BO`, async () => {
+      it('should go to catalog > products page', async function () {
+        await this.pageObjects.boBasePage.goToSubMenu(
+          this.pageObjects.boBasePage.catalogParentLink,
+          this.pageObjects.boBasePage.productsLink,
+        );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:111]:
```suggestion
        );
        
```

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        numberOfProducts = await this.pageObjects.productsPage.resetAndGetNumberOfLines();
+        await expect(numberOfProducts).to.be.above(0);
+      });
+
+      it('should create product and check the products number', async function () {
+        await this.pageObjects.productsPage.goToAddProductPage();
+        const createProductMessage = await this.pageObjects.addProductPage.createEditProduct(
+          test.args.productToCreate,
+          test.args.enabled,
+        );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:128]:
```suggestion
        );
        
```

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        await expect(createProductMessage).to.equal(this.pageObjects.addProductPage.settingUpdatedMessage);
+      });
+    });
+
+    describe('Check created product in monitoring page', async () => {
+      it('should go to catalog > monitoring page', async function () {
+        await this.pageObjects.addProductPage.goToSubMenu(
+          this.pageObjects.boBasePage.catalogParentLink,
+          this.pageObjects.boBasePage.monitoringLink,
+        );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:139]:
```suggestion
        );
        
```

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+        const pageTitle = await this.pageObjects.monitoringPage.getPageTitle();
+        await expect(pageTitle).to.contains(this.pageObjects.monitoringPage.pageTitle);
+        numberOfProductsIngrid = await this.pageObjects.monitoringPage.resetAndGetNumberOfLines(
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:144]:
```suggestion

        numberOfProductsIngrid = await this.pageObjects.monitoringPage.resetAndGetNumberOfLines(
```

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+          test.args.gridName,
+        );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:146]:
```suggestion
        );
        
```

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        await expect(numberOfProductsIngrid).to.be.at.least(1);
+      });
+
+      it(`should filter products ${test.args.productType} grid and check existence of new product`, async function () {
+        await this.pageObjects.monitoringPage.filterTable(
+          test.args.gridName,
+          'input',
+          'name',
+          test.args.productToCreate.name,
+        );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:157]:
```suggestion
        );
        
```

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+        const textColumn = await this.pageObjects.monitoringPage.getTextColumnFromTable(
+          test.args.gridName,
+          1,
+          'name',
+        );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:163]:
```suggestion
        );
        
```

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+    });
+
+    describe('Delete product from monitoring page', async () => {
+      it(`should filter products ${test.args.productType} grid`, async function () {
+        await this.pageObjects.monitoringPage.filterTable(
+          test.args.gridName,
+          'input',
+          'name',
+          test.args.productToCreate.name,
+        );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:181]:
```suggestion
        );
        
```

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+        const textColumn = await this.pageObjects.monitoringPage.getTextColumnFromTable(
+          test.args.gridName,
+          1,
+          'name',
+        );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:187]:
```suggestion
        );
        
```

**Review by NeOMakinG — CHANGES_REQUESTED** [review]

File: tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/catalog/monitoring/02_monitoringProducts.js:103]: done

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**